Java/core/src/main/java/com/amazon/randomcutforest/store/IndexManager.java

```diff
@@ -30,54 +31,110 @@
 public class IndexManager {
 
     protected final int capacity;
-    protected final int[] freeIndexes;
+    protected int[] freeIndexes;
     protected int freeIndexPointer;
     protected final BitSet occupied;
 
     /**
-     * Create a new store with the given capacity.
+     * Create a new indexmanager with the given capacity.
      * 
      * @param capacity The total number of values that can be saved in this store.
      */
     public IndexManager(int capacity) {
-        this(capacity, new BitSet(capacity));
+        this(capacity, null);
```

Comment:
Can we change this to
```
this(capcity, new BitSet(capacity));
```
instead of checking for null in `IndexManager(int, BitSet)`?

Reply:
Done - merging with the other indexManager.